common/Services/PackageDeploymentService.cs

```diff
@@ -31,4 +34,49 @@ public async Task RegisterPackageForCurrentUserAsync(string packageFamilyName, R
             throw new RegisterPackageException(result.ErrorText, result.ExtendedErrorCode);
         }
     }
+
+    /// <inheritdoc />
+    public IEnumerable<Package> FindPackagesForCurrentUser(string packageFamilyName, Version minVersion1, Version version1Ceiling, Version minVersion2)

```

Comment:
Great candidate method in this file!! Thanks!

Suggestion: Since now this is reusable, I suggest updating the signature so we can pass flexible version ranges. Something like this maybe?:
```cs
public IEnumerable<Package> FindPackagesForCurrentUser(string packageFamilyName, params (Version minVersion, Version maxVersion)[] ranges)
```

where:
```cs
FindPackagesForCurrentUser(string packageFamilyName)                 // Finds all matches
FindPackagesForCurrentUser(string packageFamilyName, (a, b))         // Finds all matches where version in {a..b}
FindPackagesForCurrentUser(string packageFamilyName, (a, b), (c, d)) // Finds all matches where version in {a..b} or version in {c..d}
```

Or (more flexible usage) we can even have a lambda condition as parameter if we want to create a VersionHelper class:
```cs
FindPackagesForCurrentUser(string packageFamilyName, (version) => VersionHelper.IsBetween(version, a, b));
```